numba/typing/templates.py

```diff
@@ -627,7 +645,7 @@ def method_impl(context, builder, sig, args):
             call = context.get_function(disp_type, sig)
             # Link dependent library
             context.add_linking_libs(getattr(call, 'libs', ()))
-            return call(builder, args)
+            return call(builder, _adjust_omitted_args(sig.args, args))
```

Comment:
Out of scope so new ticket added... this needs enhancing to also do validation https://github.com/numba/numba/issues/3708, it was too easy to create invalid overloads when testing this.